proto/proto/store.proto

```diff
@@ -47,7 +53,7 @@ service Rpc {
     //
     // A basic note sync can be implemented by repeatedly requesting the previous response's block until reaching the
     // tip of the chain. 
-    rpc SyncNotes(requests.SyncNoteRequest) returns (responses.SyncNoteResponse) {}
+    rpc SyncNotes(NoteTags) returns (SyncNotesResponse) {}
```

Comment:
I would probably keep `SyncNoteRequest` here because the underlying message contains more than just tags.

Also, maybe not for this PR, but we should document how many tags can be passed in and adjust if needed. For example, I think currently the limit may be something like 1000, but we probably need to restrict it quite a bit more. This applies to most endpoints which take lists as parameters.

Reply:
I created #1080 to track the second.